expired sessions - errors on deletion

When a session is expired it still shows using `./fcli ssc session ls`.
Trying to delete the expired sessions by name. e.g. `./fcli ssc session logout proxySetSession -u <name> -p <pass>` 

this returns to stdout
```
java.lang.IllegalStateException: Error reading session descriptor, please try logging in again
        at com.fortify.cli.common.session.helper.AbstractSessionHelper.lambda$get$0(AbstractSessionHelper.java:51)
        at com.fortify.cli.common.session.helper.AbstractSessionHelper.conditionalThrow(AbstractSessionHelper.java:124)
        at com.fortify.cli.common.session.helper.AbstractSessionHelper.get(AbstractSessionHelper.java:51)
        at com.fortify.cli.common.session.cli.cmd.AbstractSessionLogoutCommand.getJsonNode(AbstractSessionLogoutCommand.java:33)
        at com.fortify.cli.common.output.cli.cmd.AbstractOutputCommand.run(AbstractOutputCommand.java:33)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2104)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2539)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2531)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2493)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2351)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2495)
        at picocli.CommandLine.execute(CommandLine.java:2248)
        at com.fortify.cli.app.FortifyCLI.execute(FortifyCLI.java:74)
        at com.fortify.cli.app.FortifyCLI.main(FortifyCLI.java:56)
Caused by: java.lang.IllegalStateException: SSC session 'proxySetSession' cannot be retrieved or has expired, please login again
        at com.fortify.cli.common.session.helper.AbstractSessionHelper.checkNonExpiredSessionAvailable(AbstractSessionHelper.java:130)
        at com.fortify.cli.common.session.helper.AbstractSessionHelper.get(AbstractSessionHelper.java:47)
        ... 11 more
```
when I then run `ls` command again, the deleted session is gone.

Shoul the user be receiving these errors? It does not seem like a defect but as a negative test I don't think this should be printed to stdout.

--- a/fcli-core/fcli-common/src/main/java/com/fortify/cli/common/session/cli/cmd/AbstractSessionLogoutCommand.java
+++ b/fcli-core/fcli-common/src/main/java/com/fortify/cli/common/session/cli/cmd/AbstractSessionLogoutCommand.java
@@ -30,8 +30,13 @@ public abstract class AbstractSessionLogoutCommand<D extends ISessionDescriptor>
         var sessionHelper = getSessionHelper();
         if ( sessionHelper.exists(sessionName) ) {
         	result = sessionHelper.sessionSummaryAsObjectNode(sessionName);
-            logout(sessionName, sessionHelper.get(sessionName, false));
-            getSessionHelper().destroy(sessionName);
+            try {
+                logout(sessionName, sessionHelper.get(sessionName, false));
+            } catch (Exception e){
+                throw e;
+            } finally {
+                getSessionHelper().destroy(sessionName);
+            }
         }
         return result;
     }

--- a/fcli-core/fcli-common/src/main/java/com/fortify/cli/common/session/cli/cmd/AbstractSessionLogoutCommand.java
+++ b/fcli-core/fcli-common/src/main/java/com/fortify/cli/common/session/cli/cmd/AbstractSessionLogoutCommand.java
@@ -30,8 +30,7 @@ public abstract class AbstractSessionLogoutCommand<D extends ISessionDescriptor>
         var sessionHelper = getSessionHelper();
         if ( sessionHelper.exists(sessionName) ) {
         	result = sessionHelper.sessionSummaryAsObjectNode(sessionName);
-            logout(sessionName, sessionHelper.get(sessionName, true));
-            // TODO Optionally delete all variables
+            logout(sessionName, sessionHelper.get(sessionName, false));
             getSessionHelper().destroy(sessionName);
         }
         return result;
@@ -46,6 +45,10 @@ public abstract class AbstractSessionLogoutCommand<D extends ISessionDescriptor>
     public boolean isSingular() {
     	return false;
     }
-    
+
+    /*******************************************************************************
+    * This method will always be invoked on existing sessions, independent of whether the session has expired
+    * This is to ensure cleanup of the local session directory and tokens stored in ssc (if the token has already been cleaned up by ssc this should not result in an error)
+    *******************************************************************************/
     protected abstract void logout(String sessionName, D sessionDescriptor);
 }

--- a/fcli-core/fcli-ssc/src/main/java/com/fortify/cli/ssc/token/helper/SSCTokenConverter.java
+++ b/fcli-core/fcli-ssc/src/main/java/com/fortify/cli/ssc/token/helper/SSCTokenConverter.java
@@ -13,10 +13,13 @@
 package com.fortify.cli.ssc.token.helper;
 
 import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
 
 import org.apache.commons.codec.binary.Base64;
 
+
 public final class SSCTokenConverter {
+    private static Pattern applicationTokenPattern = Pattern.compile("^[\\da-f]{8}(?:-[\\da-f]{4}){3}-[\\da-f]{12}$");
     private SSCTokenConverter() {}
     
     public static final String toApplicationToken(String token) {
@@ -36,11 +39,18 @@ public final class SSCTokenConverter {
     }
     
     public static final boolean isApplicationToken(String token) {
-        return token.matches("^[\\da-f]{8}(?:-[\\da-f]{4}){3}-[\\da-f]{12}$");
+        return applicationTokenPattern.matcher(token).matches();
     }
     
     private static final String decode(String token) {
-        return new String(Base64.decodeBase64(token), StandardCharsets.UTF_8);
+        return validateApplicationTokenFormat(new String(Base64.decodeBase64(token), StandardCharsets.UTF_8));
+    }
+    
+    private static final String validateApplicationTokenFormat(String token) {
+        if(!isApplicationToken(token)) {
+            throw new IllegalArgumentException("The provided token could not be decoded to a valid application token format");
+        }
+        return token;
     }
     
     private static final String encode(String token) {

--- a/fcli-core/fcli-ssc/src/main/java/com/fortify/cli/ssc/token/helper/SSCTokenConverter.java
+++ b/fcli-core/fcli-ssc/src/main/java/com/fortify/cli/ssc/token/helper/SSCTokenConverter.java
@@ -31,7 +31,7 @@ public final class SSCTokenConverter {
     }
     
     public static final String toRestToken(String token) {
-        return isApplicationToken(token) ? encode(token) : token; 
+        return isApplicationToken(token) ? encode(token) : validateRestTokenFormat(token); 
     }
     
     public static final char[] toRestToken(char[] token) {
@@ -50,6 +50,11 @@ public final class SSCTokenConverter {
         if(!isApplicationToken(token)) {
             throw new IllegalArgumentException("The provided token could not be decoded to a valid application token format");
         }
+        return token;
+    }
+    
+    private static final String validateRestTokenFormat(String token) {
+        decode(token);
         return token;
     }
     

